### DISCUSSION TITLE
SIP needs Python 3.

### DISCUSSION THREAD

**Discussion by Dan-in-CA** [Dan-in-CA/SIP#291] [Announcements]
During a resent update to SIP's plugin manager I used some Python3 code. As a result the plugin manager which is itself a plugin will not load under Python2.
Because Python 2 has reached end of life and the latest Pi OS (formally Raspbian) now has only Python3 installed SIP should be considered Python3-only compatible. There will soon be a Python3-only version of SIP released.

**Comment by alanwbaker** [comment]
Hi Dan,

First, thank you for this excellent software. I've used it for years and I love it.

I installed Bullseye on my Pi and had to reinstall SIP. I used Python 3 and followed the steps in your very clear instruction guide. After rebooting, I can access the web UI, but found that the service did not start automatically. I checked the installation and

- sip3.service is in etc/systemd/system
- python3-rpi.gpio is installed
- sudo systemctl enable sip3.service ran with no message
- sudo service sip3 returns message sip3: unrecognized service

Can you give me a clue?
Alan

**Reply by Dan-in-CA** [reply]
This can happen if SIP is installed in a different directory from /home/pi/.
If your installation has a different user name or if SIP is installed in a different directory you would need to edit either sip.service or sip3.service to change the path included in the *.service file. 
From the SIP directory you could use nano sip3.service.

When you run "sudo systemctl enable sip3.service" you should get a message.

Dan

**Reply by alanwbaker** [reply]
I couldn't figure out the issue with systemd, so I started it from /etc/rc.local. 

When I do Run Once, the software works just fine, but the physical stations don't start. In your instructions you warned that this will happen if python3-rpi.gpio is not installed, so I verified that the latest python3-rpi.gpio is installed. Any suggestions?

Alan

**Reply by alanwbaker** [reply]
Sorry, Dan. It was a power problem. 😉

Alan
